### PR TITLE
BnListbox: Handle objects and native form submission

### PR DESCRIPTION
**Pull request by gmq**
## Context
Banano is a component library meant for use in Platanus projects. BnListbox acts as a select replacement, using the Listbox component from the library `HeadlessUI`. When handling objects, the `Listbox` component receives and returns the whole selected object(s) which is different from what we normally use (ids) which may cause issues for adoption.

## This PR

This PR adds an optional wrapper function around Listbox that parses and returns the expected object value (usually the id), however the old way of handling objects is kept as a setting.

In addition, a hidden input is added to allow the component to work with native forms (especifically Rails).

### CODE REVIEW

**Comment by gmq** [comment]
[Listbox debe funcionar con rails](https://trello.com/c/64jHOJoD/16-listbox-debe-funcionar-con-rails)

**Comment by gmq** [comment]
[En Listbox, permitir que, al usar objetos como value, se pueda bindear una de sus propiedades como el value del select](https://trello.com/c/lExJbc15/29-en-listbox-permitir-que-al-usar-objetos-como-value-se-pueda-bindear-una-de-sus-propiedades-como-el-value-del-select)

File: src/components/BnListbox/BnListbox.spec.ts [hunk 1]
```diff
@@ -180,4 +180,27 @@ describe('BnListbox', () => {
       expect(button.classes().includes('bn-listbox__button--error')).toBe(true);
     });
   });
+
+  describe('hidden inputs', () => {
```
Comment (fprebolledo) [inline src/components/BnListbox/BnListbox.spec.ts:180]: Maybe you could add some context to each it in this test.  Like `with one option value` or `with multiples option values`

File: src/components/BnListbox/BnListbox.spec.ts [hunk 2]
```diff
@@ -119,7 +119,6 @@ describe('BnListbox', () => {
     wrapper.findComponent(ListboxOption).trigger('click');
     wrapper.find('form').trigger('submit');
 
-    await flushPromises();
```
Comment (fprebolledo) [inline src/components/BnListbox/BnListbox.spec.ts:116]: How do you know when to use this helper?

**Review by fprebolledo — APPROVED** [review]
It's looks very good. I left a little comment to improve the description of one test and a question. 💎

File: src/components/BnListbox/BnListbox.spec.ts (hunk 2 above)
Comment (gmq) [reply src/components/BnListbox/BnListbox.spec.ts:116]: I don't remember what made me use it in the first place, but the documentation mentions that it's for promises that are not handled directly by Vue (like http requests)

https://test-utils.vuejs.org/guide/advanced/async-suspense.html#resolving-other-asynchronous-behavior

**Pull request merged**